### PR TITLE
Add GIT_SYNC_ROOT change for non-root user in docs/ssh

### PR DESCRIPTION
**Pull request by shubhamc183**
https://github.com/kubernetes/git-sync/pull/97\#issuecomment-800606819

### CODE REVIEW

**Comment by shubhamc183** [comment]
@thockin 
https://github.com/kubernetes/git-sync/pull/97#issuecomment-800606819

**Comment by shubhamc183** [comment]
I signed it

**Comment by shubhamc183** [comment]
"I signed it"

File: docs/ssh.md [hunk 1]
```diff
@@ -106,7 +106,7 @@ that this is a Pod-wide setting, unlike the container `securityContext` above.
 If you want git-sync to run as a different (non-root) UID and GID, you can
 change these last blocks to any UID/GID you like.  SSH demands that the current
 UID be present in /etc/passwd, so in this case you will need to add the
-`--add-user` flag to git-sync's args array.
+`--add-user` flag to git-sync's args array. Also, you need to change `GIT_SYNC_ROOT` to some other location, say `/workspace`, instead of `$HOME/git` as the `$HOME` will be `/` for the user and, `GIT_SYNC_ROOT` will end up in `//tmp` which do not exist.
```
Comment (thockin) [inline docs/ssh.md:109]: Super nit-pick - can you please linewrap to match the rest? :)

Comment (thockin) [reply docs/ssh.md:109]: Also "`//tmp` which do not exist" -> "`//git` which the non-root user can't create"

**Review by thockin — COMMENTED** [review]
Also needs you to sign CLA.  Thanks!

File: docs/ssh.md [hunk 2]
```diff
@@ -106,7 +106,10 @@ that this is a Pod-wide setting, unlike the container `securityContext` above.
 If you want git-sync to run as a different (non-root) UID and GID, you can
 change these last blocks to any UID/GID you like.  SSH demands that the current
 UID be present in /etc/passwd, so in this case you will need to add the
-`--add-user` flag to git-sync's args array. Also, you need to change `GIT_SYNC_ROOT` to some other location, say `/workspace`, instead of `$HOME/git` as the `$HOME` will be `/` for the user and, `GIT_SYNC_ROOT` will end up in `//tmp` which do not exist.
+`--add-user` flag to git-sync's args array. Also, you need to change
+`GIT_SYNC_ROOT` to some other location, say `/workspace`, instead of
+`$HOME/git` as the `$HOME` will be `/` for the user and, `GIT_SYNC_ROOT` will
+end up in `//git` which do not exist.
```
Comment (thockin) [inline docs/ssh.md:112]: last pick:  s/do/does

Comment (shubhamc183) [reply docs/ssh.md]: Sorry, I did not understand this.

Comment (thockin) [reply docs/ssh.md]: sorry, - please change "do" to "does" (blame English)

Comment (shubhamc183) [reply docs/ssh.md]: Actually, I changed it to "//git which the non-root user can't create" as suggested by you previously.

Also, I created an account in Linux Foundation ID Portal account with the correct same e-mail address but didn't get the email agreement to sign for CLA.

Comment (thockin) [reply docs/ssh.md]: @caniszczyk - how can we poke the CLA machinery to check this?

File: docs/ssh.md [hunk 3]
```diff
@@ -106,7 +106,10 @@ that this is a Pod-wide setting, unlike the container `securityContext` above.
 If you want git-sync to run as a different (non-root) UID and GID, you can
 change these last blocks to any UID/GID you like.  SSH demands that the current
 UID be present in /etc/passwd, so in this case you will need to add the
-`--add-user` flag to git-sync's args array.
+`--add-user` flag to git-sync's args array. Also, you need to change
+`GIT_SYNC_ROOT` to some other location, say `/workspace`, instead of
+`$HOME/git` as the `$HOME` will be `/` for the user and, `GIT_SYNC_ROOT` will
```
Comment (thockin) [inline docs/ssh.md:111]: While we wait for CLA - the `,` in this sentence is not needed :)

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

File: docs/ssh.md (hunk 2 above)
Comment (thockin) [reply docs/ssh.md]: Are you sure it was the same email you used for this commit (sh...183@gmail.com) ?

Check your spam and make sure?  We're stuck until it is resolved. :(

**Comment by shubhamc183** [comment]
I have used the same username and email.

I didn't get the email agreement to sign checked even in my Spam and other folders.

image

**Comment by thockin** [comment]
I'm told you can open a ticket at https://support.linuxfoundation.org/ for CLA issues.

**Pull request merged**